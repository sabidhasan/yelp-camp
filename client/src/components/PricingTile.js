import React from 'react'
import PropTypes from 'prop-types'

const PricingTile = (props) => (
  <React.Fragment>
    <h3>{props.type}</h3>
    <div className='Sidebar__text'>
      {props.data.prices ? (props.data.prices[props.type] ? props.data.prices[props.type].join(' - $') : 'None')
      : `No ${props.type} pricing provided`}
    </div>
  </React.Fragment>
)

PricingTile.propTypes = {
  data: PropTypes.object,
  type: PropTypes.string.isRequired
}

export default PricingTile
