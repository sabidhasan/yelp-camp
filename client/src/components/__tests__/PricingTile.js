import React from 'react'
import toJson from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import PricingTile from '../PricingTile'

configure({ adapter: new Adapter() });

describe('<PricingTile />', () => {
  let component;
  let props;
  const type = 'Title'

  beforeEach(() => {
    props = {
      data: {prices: null},
      type
    }
    component = shallow(<PricingTile { ...props } />)
  })

  it('Component should be defined', () => {
    expect(component).toBeDefined();
  });

  it('Header element shows heading', () => {
    expect(component.find('h3').text()).toBe('Title');
  });

  it('Sidebar shows error message when no pricing provided', () => {
    expect(component.find('.Sidebar__text').text()).toBe(`No ${type} pricing provided`);
  });

  it('If pricing of type not provided, then it shows None', () => {
    props = {
      data: {prices: {Title: null}},
      type
    }
    component = shallow(<PricingTile { ...props } />)
    expect(component.find('.Sidebar__text').text()).toBe('None');
  });

  it('Single price is rendered correctly', () => {
    props = {
      data: {prices: {Title: [50]}},
      type
    }
    component = shallow(<PricingTile { ...props } />)
    expect(component.find('.Sidebar__text').text()).toBe('$50');
  });

  it('Price range is rendered correctly', () => {
    props = {
      data: {prices: {Title: [50, 100]}},
      type
    }
    component = shallow(<PricingTile { ...props } />)
    expect(component.find('.Sidebar__text').text()).toBe('$50 - $100');
  });

 it('LoginForm component snapshot matches what is expected', () => {
   expect(toJson(component)).toMatchSnapshot();
 });
 
});
