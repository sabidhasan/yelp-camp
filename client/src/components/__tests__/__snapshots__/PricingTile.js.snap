// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PricingTile /> LoginForm component snapshot matches what is expected 1`] = `
<Fragment>
  <h3>
    Title
  </h3>
  <div
    className="Sidebar__text"
  >
    No Title pricing provided
  </div>
</Fragment>
`;
